src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,271 @@
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `-error` topic for failover processing.
+     * @param message
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
+                    topics = ORDER_RECEIVED_TOPIC_RETRY,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceivedRetry(org.springframework.messaging.Message<OrderReceived> message) {
+        handleMessage(message);
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC, groupId = ORDER_RECEIVED_GROUP,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceived(String orderReceivedUri)
-            throws SerializationException, ExecutionException, InterruptedException {
-        try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, ex);
+    /**
+     * Error (`-error`) topic listener/consumer is enabled when the application is launched in error
+     * mode (IS_ERROR_QUEUE_CONSUMER=true). Receives messages up to `errorRecoveryOffset` offset. Calls `handleMessage`
+     * method to process received message. If the `retryable` processor is unsuccessful with a `retryable` error, after
+     * maximum numbers of attempts allowed, the message is republished to `-retry` topic for failover processing.
+     * This listener stops accepting messages when the topic's offset reaches `errorRecoveryOffset`.
+     * @param message
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP_ERROR, groupId = ORDER_RECEIVED_GROUP_ERROR,
+                    topics = ORDER_RECEIVED_TOPIC_ERROR,
+                    autoStartup = "${uk.gov.companieshouse.item-handler.error-consumer}")
+    public void processOrderReceivedError(org.springframework.messaging.Message<OrderReceived> message) {
+        long offset = Long.parseLong("" + message.getHeaders().get("kafka_offset"));
+        if (offset <= errorRecoveryOffset) {
+            handleMessage(message);
+        }
+        else {
+            LOGGER.info(String.format("Pausing error consumer \"%1$s\" as error recovery offset '%2$d' reached.",
+                    ORDER_RECEIVED_GROUP_ERROR, errorRecoveryOffset));
+            registry.getListenerContainer(ORDER_RECEIVED_GROUP_ERROR).pause();
         }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceivedRetry(String orderReceivedUri)
-            throws SerializationException, ExecutionException, InterruptedException {
+    /**
+     * Handles processing of received message.
```

Comment:
I think you might consider moving (or possibly duplicating) much of the explanation of the retry logic from the `processOrderReceived*` method JavaDocs to the JavaDoc for this method? It is this method that implements the logic. To be more DRY, you could outline the logic in this JavaDoc then possibly refer to it from the JavaDocs of those methods?

Reply:
This has been updated. Thanks